--- cfl-0.5.0/configure.in.orig	Sun Feb 23 12:01:53 2003
+++ cfl-0.5.0/configure.in	Sun Feb 23 12:05:04 2003
@@ -64,7 +64,16 @@
 dnl Checks for header files.
 AC_HEADER_STDC
 AC_CHECK_HEADERS([stdlib.h string.h unistd.h])
-AC_CHECK_HEADER(gdsl.h,	[], AC_CHECK_HEADER(gdsl/gdsl.h, [CFLAGS="-I/usr/local/include/gdsl $CFLAGS"], AC_MSG_ERROR([Cannot find GDSL header.])))
+AC_CHECK_HEADER(gdsl.h,        ,
+               [ AC_CHECK_HEADER(gdsl/gdsl.h, [
+                                               AC_CHECK_LIB(gdsl,gdsl_get_version_string,
+                                                            CFLAGS="$CFLAGS -I/usr/local/include/gdsl -I/usr/include/gdsl"
+                                                            LIBS="$LIBS -lgdsl",
+                                                            AC_MSG_ERROR(Cannot find GDSL.))
+                                               ], [ AC_MSG_ERROR(Cannot find GDSL.)]
+                               )
+               ]
+               )
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
